File: Kotlin/그리디 알고리즘/1931.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main()= with(BufferedReader(InputStreamReader(System.`in`))) {
    val N = readLine().toInt()
    val arr = arrayOfNulls<Type>(N)
    var res = 0

    for (i in 0 until N){
        val (start, end) = readLine().split(' ').map { it.toInt() }
        arr[i] = Type(start, end)
    }

    arr.sortWith(compareBy({it!!.end}, {it!!.start}))
    var temp = arr[0]!!.end
    res += 1
    for (i in 1 until N){
        if (arr[i]!!.start >= temp){
            res += 1
            temp = arr[i]!!.end
        }
    }

    println(res)
}

data class Type(val start:Int, val end:Int)

File: BOJ/Python/그리디 알고리즘/1931.py
N = int(input())
arr = list()
res = 0

for _ in range(N):
    arr.append(tuple(map(int, input().split())))

arr.sort(key=lambda x: (x[1], x[0]))

end = arr[0][1]
res += 1

for i in range(1, N):
    if arr[i][0] >= end:
        res += 1
        end = arr[i][1]

print(res)
